extensions/amp-script/0.1/amp-script.js

```diff
@@ -162,13 +165,14 @@ export class AmpScript extends AMP.BaseElement {
   /**
    * @override
    */
-  onMeasureChanged() {
-    if (this.layoutCompleted_) {
+  onLayoutMeasure() {
+    if (this.layoutCompleted_ || this.reportedZeroSize_) {
       return;
     }
 
     const {width, height} = this.getLayoutSize();
     if (width === 0 && height === 0) {
+      this.reportedZeroSize_ = true;
```

Comment:
Is this so that `<amp-script>` doesn't double-report? If yes can you add a small extra unit test (like the ones you've modified) checking: `it('should not warn twice'`

Reply:
TBH, I kind of felt it a bit silly to overtest this part. I think the only role it plays is in avoiding spamming the console. And TBH we never report this sort of thing, so not sure why we are doing it here. I kind of rather we remove it completely.

Reply:
The reason it is helpful today is as a bandaid because someone can make `layout=container` amp-script that initialized with 0 size. Right now, its pretty bad that it may never layout, so we warn the dev. Ideally `BaseElement` could have an overridable method for `shouldAlwaysLayout`, or we special-case amp-script somehow.  (this is as opposed to something like amp-bind/amp-state which never have a UI and so we can always make fixed/hidden with nonzero size)

Reply:
`shouldAlwaysLayout` does sounds like a reasonable feature.